clients/js/test/offline.test.ts

```diff
@@ -0,0 +1,15 @@
+import { expect, test } from "@jest/globals";
+import { ChromaClient } from "../src/ChromaClient";
+
+test("it fails with a nice error", async () => {
+  const chroma = new ChromaClient({ path: "http://example.invalid" });
+  try {
+    await chroma.createCollection({ name: "test" });
+    throw new Error("Should have thrown an error.");
+  } catch (e) {
+    expect(e instanceof Error).toBe(true);
+    expect((e as Error).message).toMatchInlineSnapshot(
+      `"Failed to connect to chromadb. Make sure your server is running and try again. If you are running from a browser, make sure that your chromadb instance is configured to allow requests from the current origin using the CHROMA_SERVER_CORS_ALLOW_ORIGINS environment variable."`
```

Comment:
This is super nice! 

A frequent problem I see is JS sucks with dual-stack systems where most people will connect to localhost, which maps to :1; however, their server is listening on 127.0.0.1. Can we force JS to try both stacks before giving up?

Reply:
That is an interesting question. My gut reaction is that it might cause some surprising behaviors in certain circumstances (e.g. the user has something else running at :1:8000 which causes some cryptic errors). 

I think that would be a good conversation to have but I think adding that functionality would be outside of the original scope of this PR.